core/lib/instrument.c

```diff
@@ -4881,6 +4881,20 @@ dr_raw_tls_cfree(uint offset, uint num_slots)
     return os_tls_cfree(offset, num_slots);
 }
 
+DR_API
+void
+dr_raw_write_tls_slot(reg_id_t *tls_register, ushort offset, void *value)
+{
+    d_r_set_tls(offset, value);
```

Comment:
For safety I think this should do some checking in debug build and warn the client if it tries to clobber DR slots.

Reply:
If tls_register == SEG_TLS and offset == TLS_REG[0123]_SLOT ?